main.go

```diff
@@ -137,8 +137,8 @@ func getConfig() (*config, error) {
 
 func getCustomLoggerFromContext(ctx *gin.Context) *logrus.Entry {
 	deliveryID, ok := ctx.Get("delivery")
-	if !ok {
-		return nil
+	if !ok || deliveryID == nil {
+		return logrus.WithField("delivery", "nil")
```

Comment:
I don't think that this will fix the issue. As you said judging  by the docs it won't be `nil` unless `ok` if false. My theory is that `deliveryID` is _something_ but not an string. Then passing it down the road at some point breaks.

Maybe we can check if it's an string, and when not do this `return logrus.WithField("delivery", "nil")` - which also will work in the nil case.